### ISSUE TITLE
ImportResolver not respecting py.typed packages (PEP 561)

### ISSUE THREAD

**Issue by ian-r-rose** [microsoft/pyright#1764]
**Describe the bug**

Opening this following some discussion in #1761. Since PEP 561 python packages can declare themselves as having inline type annotations by including a file `py.typed`. Annotations from such packages are supposed to have higher precedence than ones from `typeshed` (the theory being that library authors should have the opportunity to declare their own type annotations). This ordering is also described in the docs for [this project](https://github.com/microsoft/pyright/blob/master/docs/import-resolution.md#resolution-order).

However, the current implementation of `ImportResolver` ultimately chooses type annotations from its vendored copy of `typeshed`, rather than from packages declaring themselves as `py.typed`. An example of this (as of `pyright@1.1.130`) is the `tornado` package, which has its own inline type annotations, which are more accurate than the `typeshed` ones. So, for instance, `tornado.concurrent.Future`, which is just a re-export of `asyncio.Future` (cf. #1662), winds up missing some crucial annotations.

A workaround is to point pyright at a custom `typeshed` fork, but this is difficult to distribute, and is contrary to the intent of `py.typed` (i.e., library authors get to declare their own annotations).

**To Reproduce**

```python
from tornado.concurrent import Future

async def foo():
    fut = Future()
    await fut   # current typeshed annotations are python 2 only, and missing __await__
```

**Expected behavior**

A python package should be able to declare itself `py.typed` and have that work, even if there are alternative stubs in `typeshed`, as per PEP 561

**VS Code extension or command-line**

Command-line `pyright@1.1.130`

**Comment by jakebailey** [comment]
The import resolver has been changed to prefer the `py.typed` library over typeshed, so I believe this to be fixed in the next version.

**Comment by jakebailey** [comment]
FWIW I tested this out with a number of `py.typed` libraries (tornado, PyJWT, the all-new Flask and deps), and it worked as expected.

**Comment by ian-r-rose** [comment]
Thanks for the update! Closing as resolved

**Comment by jakebailey** [comment]
We typically leave issues open until the fix is released; reopening.

**Comment by erictraut** [comment]
This is included in pyright 1.1.139, which I've just published. It will also be included in the next release of pylance.
